### ISSUE TITLE
Workflow `deploy` broke

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#5874]
Workflow `deploy` on ubuntu-20.04 broken since 2022-04-08
- https://github.com/agda/agda/runs/5883429340?check_suite_focus=true

```
Linking /home/runner/work/agda/agda/dist-newstyle/build/x86_64-linux/ghc-9.2.2/Agda-2.6.3/build/agda-mode/agda-mode ...
/usr/local/.ghcup/ghc/9.2.2/lib/ghc-9.2.2/process-1.6.13.2/libHSprocess-1.6.13.2.a(fork_exec.o):fork_exec.c:function do_spawn_fork: 
warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking 
...
/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libpthread.a(pthread_join_common.o)(.note.stapsdt+0x14): 
error: relocation refers to local symbol "" [2], which is defined in a discarded section
  section group signature: "(null)"
```

- Last successful run:
  Environment: ubuntu-20.04
  Version: 20220330.0
  Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220330.0/images/linux/Ubuntu2004-Readme.md
  Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220330.0

- First broken run:
  Environment: ubuntu-20.04
  Version: 20220405.4
  Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220405.4/images/linux/Ubuntu2004-Readme.md
  Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220405.4

Comparing these versions: https://github.com/actions/virtual-environments/compare/ubuntu20/20220330.0...ubuntu20/20220405.4
- Ubuntu LTS: 20.04.3 -> 20.04.4
- GNU C++ 9: 9.3.0 -> 9.4.0
- GHC: 9.2.1 -> 9.2.2
- GHCup: 0.1.17.4 -> 0.1.17.6

**Comment by andreasabel** [comment]
Problem reported to `ghcup`:
- https://gitlab.haskell.org/haskell/ghcup-hs/-/issues/356

**Comment by andreasabel** [comment]
The issue was moved to GHC:
- https://gitlab.haskell.org/ghc/ghc/-/issues/21454

There it is suggested by @bgamari that building a statically linked executable might be problematic:
https://github.com/agda/agda/blob/a1575bc63bd984c19cab97b1137771cf48a0b5dd/src/github/workflows/deploy.yml#L37

@hasufell suggests building on Alpine Linux instead, using a workflow akin to https://github.com/hasufell/stack2cabal/blob/6f12cea1bffaa804430317b1d66ce2d60c7b05b2/.github/workflows/release.yaml#L31-L132 .

**Comment by andreasabel** [comment]
Resolved (for now) by reverting to `ubuntu-18.04` in  7bda16229ef4728761a9b538f17bb63e20777618.
